packages/ui/fe-bundle/src/utils/theme/loadDocsTypography.ts

```diff
@@ -0,0 +1,122 @@
+import type * as FernRegistryDocsRead from "@fern-fern/registry-browser/api/resources/docs/resources/v1/resources/read";
+
+export function getFontExtension(url: URL): string {
+    const ext = url.pathname.split(".").pop();
+    if (ext == null) {
+        throw new Error("No extension found");
+    }
+    return ext;
+}
+
+export type FontType = "headingsFont" | "bodyFont" | "codeFont";
+
+export interface GenerationFontConfig {
+    fontType: FontType;
+    fontName: string;
+    fontExtension: string;
+    fontUrl: FernRegistryDocsRead.Url | undefined;
+}
+
+export function getFontConfig(
+    docsDefinition: FernRegistryDocsRead.DocsDefinition,
+    fontType: FontType
+): GenerationFontConfig | undefined {
+    const typographyFontType = docsDefinition.config.typography?.[fontType];
+    if (typographyFontType == null) {
+        return undefined;
+    }
+    const fontName = typographyFontType.name;
+    const fontUrl = docsDefinition.files[typographyFontType.fontFile];
+    const fontExtension = getFontExtension(new URL(fontUrl ?? ""));
```

Comment:
`fontUrl ?? ""` is gonna cause some confusing issues if `fontUrl` is undefined. Can we add an explicit check + error message